src/controllers/Controller.java

```diff
@@ -54,107 +51,102 @@ public void initialize(URL location, ResourceBundle resources) {
+
+        for (FoodItemModel foodItemModel : foodItemModelArrayList) {
+            final Button tempButton = new Button(foodItemModel.getName());
+            tempButton.setMinHeight(69);
+            tempButton.setMinWidth(159);
 
-        for (FoodItemModel foodItemModel : foodItemsList) {
+            foodItemPane.getChildren().addAll(tempButton);
+            tempButton.setOnAction(new EventHandler<ActionEvent>() {
+                @Override
+                public void handle(ActionEvent event) {
+                    String selectedItem = tempButton.getText();
 
-            observableList.add(foodItemModel.getName());
+                    addItemToReceipt(selectedItem, selectedCategory, categoryModelHashMap);
+                }
+            });
         }
-        foodItemsListView.setItems(observableList);
     }
 
-    private void initializeReceiptTable() {
-
-        foodItemsListView.setOnMouseClicked(new EventHandler<MouseEvent>() {
-            @Override
-            public void handle(MouseEvent mouseEvent) {
+    private void addItemToReceipt(String selectedItem, String categoryName, HashMap<String, CategoryModel> categoryModelHashMap) {
 
-                String selectedFoodName = foodItemsListView.getSelectionModel().getSelectedItem().toString();
-                String categoryName = categoryListView.getSelectionModel().getSelectedItem().toString();
-                CategoryModel categoryModel = categoryModelHashMap.get(categoryName);
+        FoodItemsService foodItemsService = new FoodItemsService();
+        ArrayList<FoodItemModel> foodItemModelArrayList = foodItemsService.get(categoryName);
+        CategoryModel categoryModel = categoryModelHashMap.get(categoryName);
 
-                ArrayList<FoodItemModel> foodItem = categoryModel.getFoodItemsList();
-                for (FoodItemModel foodItemModel : foodItem) {
-                    if (selectedFoodName.equals(foodItemModel.getName())) {
+        for (FoodItemModel foodItemModel : foodItemModelArrayList) {
+            if (selectedItem.equals(foodItemModel.getName())) {
 
-                        FoodItemModel foodItemsList = categoryModel.find(selectedFoodName);
-                        selectedFoodItemsToDisplay.addAll(foodItemsList);
+                FoodItemModel foodItemsList = categoryModel.find(selectedItem);
+                selectedFoodItemsToDisplay.addAll(foodItemsList);
 
-                        Double individualItemPrice = Double.valueOf(foodItemModel.getPrice());
-                        receipt.updateTotal(individualItemPrice);
-                    }
-                }
-                taxField.setText(String.valueOf(receipt.getTax()));
-                totalField.setText(String.valueOf(receipt.getGrandTotal()));
-                receiptTableView.setItems(selectedFoodItemsToDisplay);
+                Double individualItemPrice = Double.valueOf(foodItemModel.getPrice());
+                receipt.updateTotal(individualItemPrice);
```

Comment:
wait it looks like you are doing that a couple lines lower. (calling grand total on the receipt and not passing in any params.)  Now here just add the food item to the receipt's list of items purchased and move the getting of the price and the adding to a grand total var into the grandTotal function so that it calculates on the fly.